### ISSUE TITLE
Can't filter on array of arrays

### ISSUE THREAD

**Issue by sbrooymans** [json-path/JsonPath#523]
Hello guys,

I have a JSON like this :
{
  "_source": {
    "Asset": [
      {
        "Ref": [
          {
            "string": [
              "1358420"
            ],
            "@field": "REF"
          }
        ],
        "ImageType": [
          [
            {
              "string": "Image",
              "@code": "ImageType-DAM_MTYP_IMAGE",
              "@field": "IMAGE_TYPE"
            }
          ],
          [
            {
              "string": "Packshot",
              "@code": "ImageType-DAM_MTYP_3014",
              "@field": "IMAGE_TYPE"
            }
          ]
        ]
	  },	
      {
        "Ref": [
          {
            "string": [
              "896_1"
            ],
            "@field": "REF"
          }
        ],
        "ImageType": [
          [
            {
              "string": "Product Photo",
              "@code": "ImageType-DAM_MTYP_3011",
              "@field": "IMAGE_TYPE"
            }
          ],
          [
            {
              "string": "Packshot",
              "@code": "ImageType-DAM_MTYP_3014",
              "@field": "IMAGE_TYPE"
            }
          ]
        ],
    ],
  }
}

I have to filter the assets on Ref and ImageType code.
I'm using this filter for the Ref :
Filter refFilter = Filter.filter(where("Ref[0].string[0]").is("1358420"));
parse(json).read("_source.Asset[?]", refFilter);
This is working.

Now for the ImageType code :
Filter imageTypeFilter = Filter.filter(where("ImageType[0][0].@code").is("ImageType-DAM_MTYP_IMAGE"));
This is working.

However,
Filter imageTypeFilter = Filter.filter(where("ImageType[0][1].@code").is("ImageType-DAM_MTYP_3014"));
This doesn't work.
Can you tell me why please ? Is it a problem in my code or is there a bug ?
Thank you in advance for your answer. I'm struggling with this for hours...

Regards

**Comment by hf-kklein** [comment]
Could you please modify the JSON above such that it's valid? Right now it's invalid, e.g. because of the trailing commas at the end.

**Comment by sbrooymans** [comment]
Done. Sorry I tried to simplify it at the maximum. The original one is more than 11000 lines !

**Comment by hf-kklein** [comment]
I simplified it a bit more. This is similar to the object in your `"Asset"` selected by the first filter.
```js
[
   {
      "foo" : [
         [
            {
               "prop" : "a"
            }
         ],
         [
            {
               "prop" : "b",
              "i want": "this"
            }
         ]
      ]
   },
   {
      "foo" : [
         [
            {
               "prop" : "c"
            }
         ],
         [
            {
               "prop" : "d",
               "not": "this"
            }
         ]
      ]
   }
]
```

This works:
```js
$..[?(@.foo[0][0]['prop']=='a')]
```
This doesn't:
```js
$..[?(@.foo[0][1]['prop']=='b')]
```

**Comment by sbrooymans** [comment]
Yes, this is exactly my problem. Does this mean there is a bug ?

**Comment by hf-kklein** [comment]
Guess it’s a bug. I already encountered a similar problem in #472

**Comment by sbrooymans** [comment]
I assume it won't be fixed in the next release ?

**Comment by hf-kklein** [comment]
@sbrooymans in case you're not following the issues in this repo: There hasn't been a release recently but in #500 someone said there was a release in Q1 this year. But since there's no pull request on this specific issue so far, I doubt it's gonna be fixed in the near future.

**Comment by cmunilla** [comment]
Regardless of the JsonPath implementation and limitations, in these cases the appropriate indexes are [1][0] and not [0][1] for what you are searching for.
